ReactSkia/components/RSkComponentActivityIndicatorManager.cpp

```diff
@@ -0,0 +1,120 @@
+
+RSkComponentActivityIndicatorManager::RSkComponentActivityIndicatorManager(){
+  animRequest_ = new RnsJsRequestAnimation([this](double timestamp){
+    RNS_LOG_DEBUG("[" << this->animRequest_ << "] Register Activity Indicator request Animation callback [" << timestamp << "]");
+#ifdef RNS_ACTIVITY_INDICATOR_ANIMATION_THROTTLE
+    static double previousValue = timestamp;
+    if((timestamp - previousValue) > RNS_ACTIVITY_INDICATOR_ANIMATION_THROTTLE){
+      handleActivityIndicatorAnimation(timestamp);
+      previousValue = timestamp;
+    }
+#else
+    handleActivityIndicatorAnimation(timestamp);
+#endif
+  });
+}
+
+RSkComponentActivityIndicatorManager* RSkComponentActivityIndicatorManager::getActivityIndicatorManager(){
+  std::lock_guard<std::mutex> lock(mutex_);
+  if (activityIndicatorManager_ == nullptr){
+    activityIndicatorManager_ = new RSkComponentActivityIndicatorManager;
+  }
+  return activityIndicatorManager_;
+}
+
+void RSkComponentActivityIndicatorManager::addComponent(std::weak_ptr<RSkComponent> candidate){
+  std::shared_ptr<RSkComponent> component = candidate.lock();
+
+  if(component && component->layer().get()){
+    aiComponentList_.push_back(candidate);
+
+    if(aiComponentList_.size() == 1){
+      animRequest_->start();
+    }
+  } 
+}
+
+void RSkComponentActivityIndicatorManager::removeComponent(Tag tag){
+  if(aiComponentList_.size() == 0){
+    return;
+  }
+  
+  std::vector<std::weak_ptr<RSkComponent>>::iterator iter;
+
+  for(iter = aiComponentList_.begin(); iter != aiComponentList_.end(); iter++){
+    std::shared_ptr<RSkComponent> component = iter->lock();
+    if(component){
+      if(component->getComponentData().tag == tag){
+        aiComponentList_.erase(iter);
+        break;
+      }
+    }
+  }
+
+  if(aiComponentList_.size() == 0){
+    animRequest_->stop();
+  } 
+}
+
+void RSkComponentActivityIndicatorManager::handleActivityIndicatorAnimation(double timestamp){
+  if(aiComponentList_.size() == 0){
+    return;
+  }
+
+  std::vector<std::weak_ptr<RSkComponent>>::iterator iter = aiComponentList_.begin();
+  std::shared_ptr<RSkComponent> firstComponent = iter->lock();
+
+  if(firstComponent){
+    std::shared_ptr<RnsShell::Layer> layer = firstComponent->layer();
+
+    if(layer.get() != nullptr){
+      layer->client().notifyFlushBegin();
+
+      SkMatrix matrix;
+      unsigned int arcAngle;
+
+      for( auto & iter : aiComponentList_){
+        std::shared_ptr<RSkComponent> component = iter.lock();
+
+        if(component && component->layer().get() != nullptr){
+          RSkComponentActivityIndicator *aiComponent =  dynamic_cast<RSkComponentActivityIndicator*>(component.get());
+
+          arcAngle = aiComponent->getCurrentAngle();  
+          arcAngle = (arcAngle + 6 ) % 360;
+          matrix.setRotate(arcAngle, 0, 0);
+          aiComponent->setCurrentAngle(arcAngle);
+
+          component->layer()->transformMatrix = matrix;
```

Comment:
why are we overwritting transform matrix here ?. Does it mean, Activity indicator won't accept tranform property defined in APP ex: scale or translate ?

Reply:
Addressed